### ISSUE TITLE
Icon on the right side

### ISSUE THREAD

**Issue by reeichert** [Skyscanner/SkyFloatingLabelTextField#40]
I need an Icon on the right side, how can I do that??

thank you

**Comment by gergelyorosz** [comment]
Hi! This use case is not supported out of the box - however you can definitely do this by extending the control. You could either subclass `SkyFloatingLabelTextField`, and draw the icon to the right size. You can check how displaying an icon on the left side is implemented in `SkyFloatingLabelTextFieldWithIcon` (see source [here](https://github.com/Skyscanner/SkyFloatingLabelTextField/blob/master/Sources/SkyFloatingLabelTextFieldWithIcon.swift).

Alternatively, you could also try subclassing `SkyFloatingLabelTextFieldWithIcon`, and move iconLabel to the right side. You would need to override `layoutSubviews`, `textRectForBounds`, `editingRectForBounds` and `placeholderRectForBounds` to place the textfield to the right location (and the right size).

Please let me know if this helps. We designed the control so when subclassing, you can change the layout programmatically, like in this case.

**Comment by intonarumori** [comment]
Maybe we should consider accomplishing the icon placement with `UITextField`s `leftView` property.
In right-to-left context `UITextField` places `leftView` on the right, so we could achieve this with very little effort.

From the [documentation](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UITextField_Class/#//apple_ref/occ/instp/UITextField/leftView):

> You can use the left overlay view to indicate the intended behavior of the text field. For example, you might display a magnifying glass in this location to indicate that the text field is a search field. The left overlay view flips automatically in a right-to-left user interface.

**Comment by gergelyorosz** [comment]
That's a good point @intonarumori . @reeichert - would you mind letting us know if you are using the textfield with right-to-left text (e.g. Arabic or Hebrew), or with left-to-right text (e.g. English or Spanish).

Showing the icon on the other side with right-to-left, as Daniel said, should be something we could add to the control. Configuring which side the icon is displayed is also potentially something we could add by just adding an additional property to specify the icon location in `SkyFloatingLabelTextFieldWithIcon`.

**Comment by reeichert** [comment]
Thanks for reply. 
I'm using LTR text, if you can add and additional property like `iconSide = .Left` will be awesome!

**Comment by pichirichi** [comment]
I think that ignorer to support this request and keep the RTL support intact iconSide should have the following states:
.Left & .Right - Will be flipped when switching to RTL language (right will become left and vice versa)
.LeftForce & .RightForce - language orientation won't change the alignment (RTL will use the same value for alignment)

So the enum should have:
.Unspecified (whet we have today, will be flipped for RTL).
.Left
.Right
.LeftForce
.RightForce

**Comment by gergelyorosz** [comment]
@pichirichi good point on those potential values, as well as thinking of the "force" options. I'll give the implementation a go the next few days 👍

**Comment by pichirichi** [comment]
@gergelyorosz the alignment to right is already available from the RTL support, all is required is just the property and referencing it. I might have time tomorrow to look into it if you want.

**Comment by gergelyorosz** [comment]
Thanks @pichirichi . I'll try to squeeze some time in by tomorrow as well. Sounds like it will be a simple change then! (Just finishing another fix first)

**Comment by reeichert** [comment]
up

**Comment by alelevinas** [comment]
Hello everybody.

I was looking for something like this 

![passwordfield](https://cloud.githubusercontent.com/assets/12226623/21508047/621e3c6e-cc5c-11e6-9920-3b2706ae31f9.png)
A responsive icon on the right ;)

Anyone did an extension or something you can help me with. If not I'll try @gergelyorosz  suggestion.

Thanks in advance

**Comment by anitaa1990** [comment]
Hey guys, is there an option added to the library to add icon to the right instead of the left?

**Comment by chejarla-venkatesh** [comment]
There is no option to set iconImage to Right or Left. Still you can achieve this by coding.
Just make the textFiled class as **SkyFloatingLabelTextField** and add this two lines to your code

```
textField.rightViewMode = UITextFieldViewMode.always
textField.rightView = UIImageView(image: UIImage(named : "imageName"))
```
With this no need to use **SkyFloatingLabelTextFieldWithIcon**

**Comment by kwstasna** [comment]
@chejarla-venkatesh yes but if the text is too large, then the letters are above the image

**Comment by TomikeKrasnay** [comment]
Make the textFiled class as SkyFloatingLabelTextField and add this code
```
 self.textField.rightViewMode = .always
 let imageView = UIImageView()
 let image = UIImage(named: "icon")
 imageView.frame = CGRect(x: 0, y: 0, width: 20, height: 20)
 imageView.image = image
 self.textField.rightView = imageView
```

**Comment by bhanushalilinesh** [comment]
@TomikeKrasnay Thank you, your answer worked properly
